src/libs/PolicyUtils.ts

```diff
@@ -542,37 +542,35 @@ function getDefaultApprover(policy: OnyxEntry<Policy> | SearchPolicy): string {
     return policy?.approver ?? policy?.owner ?? '';
 }
 
-/**
- * Returns the accountID to whom the given expenseReport submits reports to in the given Policy.
- */
-function getSubmitToAccountID(policy: OnyxEntry<Policy> | SearchPolicy, expenseReport: OnyxEntry<Report>): number {
-    const employeeAccountID = expenseReport?.ownerAccountID ?? CONST.DEFAULT_NUMBER_ID;
-    const employeeLogin = getLoginsByAccountIDs([employeeAccountID]).at(0) ?? '';
-    const defaultApprover = getDefaultApprover(policy);
-
-    let categoryAppover;
-    let tagApprover;
-    const allTransactions = getAllReportTransactions(expenseReport?.reportID).sort((transA, transB) => (transA.created < transB.created ? -1 : 1));
+function getRuleApprovers(policy: OnyxEntry<Policy> | SearchPolicy, expenseReport: OnyxEntry<Report>) {
+    const categoryAppovers: string[] = [];
+    const tagApprovers: string[] = [];
+    const allReportTransactions = getAllSortedTransactions(expenseReport?.reportID);
 
     // Before submitting to their `submitsTo` (in a policy on Advanced Approvals), submit to category/tag approvers.
     // Category approvers are prioritized, then tag approvers.
-    for (let i = 0; i < allTransactions.length; i++) {
-        const transaction = allTransactions.at(i);
+    for (let i = 0; i < allReportTransactions.length; i++) {
+        const transaction = allReportTransactions.at(i);
         const tag = getTag(transaction);
         const category = getCategory(transaction);
-        categoryAppover = getCategoryApproverRule(policy?.rules?.approvalRules ?? [], category)?.approver;
+        const categoryAppover = getCategoryApproverRule(policy?.rules?.approvalRules ?? [], category)?.approver;
+        const tagApprover = getTagApproverRule(policy, tag)?.approver;
         if (categoryAppover) {
-            return getAccountIDsByLogins([categoryAppover]).at(0) ?? -1;
+            categoryAppovers.push(categoryAppover);
         }
 
-        if (!tagApprover && getTagApproverRule(policy, tag)?.approver) {
-            tagApprover = getTagApproverRule(policy, tag)?.approver;
+        if (tagApprover) {
+            tagApprovers.push(tagApprover);
         }
     }
 
-    if (tagApprover) {
-        return getAccountIDsByLogins([tagApprover]).at(0) ?? -1;
-    }
+    return [...categoryAppovers, ...tagApprovers];
+}
+
+function getManagerAccountID(policy: OnyxEntry<Policy> | SearchPolicy, expenseReport: OnyxEntry<Report>) {
```

Comment:
Here we're getting the NEXT manager account ID, right? Can we call this function something like `getNextManagerAccountID` & rename `getManagerAccountEmail` to something like `getNextManagerAccountEmail`?

The way I currently read these functions is that we should be getting the report's current manager account ID / email - which I don't believe is true